compiler_opt/rl/data_reader_test.py

```diff
@@ -88,23 +87,29 @@ def setUp(self):
         maximum=20)
     super().setUp()
 
-  @parameterized.named_parameters(
-      ('SequenceExampleDatasetFn',
-       data_reader.create_sequence_example_dataset_fn),
-      ('TFRecordDatasetFn', data_reader.create_tfrecord_dataset_fn))
-  def test_create_dataset_fn(self, test_fn):
-    example = _define_sequence_example(
-        self._agent_name, is_action_discrete=True)
+  def _create_sequence_example_datasource(self, example):
+    return [example.SerializeToString() for _ in range(100)]
+
+  def _create_tfrecord_datasource(self, example):
```

Comment:
**Optional**: you can a temporary object that cleans up after itself if used in a "with" statement?

for file-based sources you can use https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

for array based sources, you can just use return contextlib.nullcontext(array)

then use it via

with data_source_fn(...) as data_source:
   ... use data_source here

Reply:
get_temp_dir() does that - the test infrastructure cleans it up after a test method is executed.